OAuth/ResourceOwner/AbstractResourceOwner.php

```diff
@@ -301,9 +301,9 @@ protected function generateNonce()
     /**
      * Configure the option resolver
      *
-     * @param OptionsResolverInterface $resolver
+     * @param OptionsResolver $resolver
      */
-    protected function configureOptions(OptionsResolverInterface $resolver)
+    protected function configureOptions(OptionsResolver $resolver)
```

Comment:
@stloyd Yes this is typically for `FormType`. I didn't do it on this because it's a internal usage of custom classes made for this bundle.

In a nutshell, is a BC for HWIOAuthOverride but not for Symfony 2.3 (tests passed).

Do you want to apply the following method?

> Keep the legacy signature for 0.4, deprecated it and build a new one (setUpOtions?) Like Symfony did. Remove BC for next release (0.5 or 1.0)

If yes, two question:

 * Which new name should as use? It's already `configureOptions`. What about `setUpOptions`?
 * For the deprecated one, in which next version would you remove it? `0.5`? `1.0`? We should make a branch to remove it or create an issue about it.

Thanks.